File: iosApp/iosApp/ui/join/JoinUiState.swift
import Shared

struct JoinUiState {
    var pin: String = ""

    var nickname: String = ""

    var avatarId: Int = StudentAvatars.defaultId

    var isJoining: Bool = false

    var isSignedIn: Bool = false

    // 입장 전 방 정보(호스트 등급·별점) — PIN 6자리 입력 시 프리페치 (T081)
    var roomInfo: RoomInfo?

    var isLoadingRoomInfo: Bool = false
}

File: composeApp/src/commonMain/kotlin/org/sesacteamproject/passmate/ui/join/JoinUiState.kt
package org.sesacteamproject.passmate.ui.join

import org.sesacteamproject.passmate.component.StudentAvatars
import org.sesacteamproject.passmate.room.domain.model.RoomInfo

data class JoinUiState(
    val pin: String = "",
    val nickname: String = "",
    val avatarId: Int = StudentAvatars.DEFAULT_ID,
    val isJoining: Boolean = false,
    val isSignedIn: Boolean = false,
    // 입장 전 방 정보(호스트 등급·별점) — PIN 6자리 입력 시 프리페치 (T081)
    val roomInfo: RoomInfo? = null,
    val isLoadingRoomInfo: Boolean = false
)
